server/scriptController.js

```diff
@@ -20,7 +22,102 @@ const scriptController = {
       .catch((error) => {
         return next({
           log: `error: ${error} occured when getting script titles`,
-          message: 'error in getScriptTitles in scriptController' ,
+          message: 'error in getScriptTitles in scriptController',
+        });
+      });
+  },
+  saveScript: (req, res, next) => {
+    const MAXFILESIZE = 50 * 1024 * 1024; //50MB
+
+    // configure multer storage to upload files to the path from the .env file
+    const storage = multer.diskStorage({
+      destination: function (req, file, cb) {
+        cb(null, process.env.UPLOADPATH);
+      },
+      filename: function (req, file, cb) {
+        cb(null, Date.now() + file.originalname);
+      },
+    });
+    // limit file uploads to less than 50MB
+    const limits = { fileSize: MAXFILESIZE };
+
+    const upload = multer({ storage, limits });
+    const scriptUpload = upload.single('newScript');
```

Comment:
What's `newScript`?

Reply:
For multer you need to specify the field in your form data that you are sending over. In the upload component we create a new formData and then append to it ('newScript', file). So it's an arbitrary field name, I'm open to alternate naming if you think something else is clearer. 

From Upload.jsx
```
 // set the file as Form Data and send it to the server
    const formData = new FormData();
    formData.append('newScript', file);
```

Reply:
scriptFormField?